sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/Platform/DefaultPlatform.cs

```diff
@@ -12,9 +12,9 @@ namespace Azure.Monitor.OpenTelemetry.Exporter.Internals.Platform
 {
     internal class DefaultPlatform : IPlatform
     {
-        public string? GetEnvironmentVariable(string name) => Environment.GetEnvironmentVariable(name);
+        private readonly IDictionary _environmentVariables = Environment.GetEnvironmentVariables();
```

Comment:
Is this the correct approach? Why do we need to retain environment variables that are unrelated to the exporter?

Reply:
They aren't retained indefinitely. This is only part of the startup/initialization.

Reply:
This can throw as well

Reply:
Updated to not throw.